src/point.js

```diff
@@ -238,15 +238,23 @@ Point.prototype.ensureInsertable = function() {
 Point.prototype.insert = function(newChild) {
   var p = this.rightNormalized();
 
+  p.ensureInsertable();
+
   if (p.type === TEXT) {
-    parentNode(p.node).insertBefore(newChild, p.node.splitText(p.offset));
-  } else if (p.type === BEFORE) {
+    p = p.moveToAfter(p.node).rightNormalized();
```

Comment:
If we're inserting when a point is in a text node, why don't we want to split the text node?